api/v1beta1/machinehealthcheck_webhook.go

```diff
@@ -136,15 +136,27 @@ func (m *MachineHealthCheck) validate(old *MachineHealthCheck) error {
 		)
 	}
 
+	allErrs = append(allErrs, m.ValidateCommonFields(specPath)...)
+
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(GroupVersion.WithKind("MachineHealthCheck").GroupKind(), m.Name, allErrs)
+}
+
+// ValidateCommonFields validates UnhealthyConditions NodeStartupTimeout, MaxUnhealthy, and RemediationTemplate of the MHC.
+// It is used to help validate other types which define MachineHealthChecks such as MachineHealthCheckClass and MachineHealthCheckTopology.
+func (m *MachineHealthCheck) ValidateCommonFields(specPath *field.Path) field.ErrorList {
```

Comment:
```suggestion
func (m *MachineHealthCheck) ValidateCommonFields(fldPath *field.Path) field.ErrorList {
```

Reply:
I think it was just called specPath because it was literally "spec", let's use the more generic fldPath here